BrainPortal/app/controllers/nh_projects_controller.rb

```diff
@@ -84,12 +87,139 @@ def update #:nodoc:
 
   def show #:nodoc:
     @nh_project = find_nh_project(current_user, params[:id])
+    @can_add_license = @nh_project.creator_id == current_user.id
   end
 
   def files #:nodoc:
     @nh_project = find_nh_project(current_user, params[:id])
     @files      = @nh_project.userfiles
   end
 
+  def new_license #:nodoc:
+    @nh_project = find_nh_project(current_user, params[:id] || params[:nh_project_id], false)
+
+    unless @nh_project.creator_id == current_user.id
+      cb_error "Only owner can set licensing", :redirect  => :neurohub
+    end
+  end
+
+  def add_license #:nodoc:
+    content = params[:meta][:license_description]
+    @nh_project = find_nh_project(current_user, params[:id] || params[:nh_project_id], false)
+    cb_error 'Presently empty licenses are not allowed' if content.blank?
+    unless @nh_project.creator_id == current_user.id
+      cb_error "Only owner can set licensing", :redirect  => :neurohub
+    end
+    @nh_project = find_nh_project(current_user, params[:id], false)
+    @nh_project.create_license_file(content, current_user)
+    flash['message'] = 'A license is added. You can force users to sign multiple license agreements if needed'
+    redirect_to :action => :show
+  end
+
+  def show_license #:nodoc:
+    @nh_project = find_nh_project(current_user, params[:id], false)
+    @unsigned_licenses = current_user.o_unsigned_custom_licenses(@nh_project)
+    if @unsigned_licenses.empty?
+      if @nh_project.license_agreements.present?
+        flash[:error] = 'You already signed all license'
```

Comment:
Should you redirect and return? since all the license are signed.

Reply:
mind at the moment this situation does not occur unless users messes with navigation, reloads etc so I kept from . I can redirect to project. but while project redirect to license while licence redirect to project. IS there a loop chance due a malfuntion or fault?

Reply:
well will add not big deal

Reply:
redirected to project page